src/library/basesqltablemodel.cpp

```diff
@@ -293,7 +293,7 @@ void BaseSqlTableModel::select() {
     // should not disturb that if we are only removing tracks.
     qStableSort(rowInfo.begin(), rowInfo.end());
 
-    m_trackIdToRows.clear();
+    QMultiHash<TrackId, int> trackIdToRows;
```

Comment:
If you have a look at some rows above you will notice that m_trackIdToRows has already been cleared together with m_rowInfo. Both are cleared enclosed between beginRemoveRows() and endRemoveRows().

A new instance of m_rowInfo is built and then assigned between beginInsertRows() and endInsertRows(). The same should be done for m_trackIdToRows. If there is some special intention why m_trackIdToRows is treated differently than m_rowInfo then this has to be explained.

Both data structures belong together and should be modified (more or less) atomically. I've just pushed some debug assertions that check some of those constraints.